src/main/scala/castalia/matcher/RequestMatcherActor.scala

```diff
@@ -0,0 +1,58 @@
+package castalia.matcher
+
+import akka.actor._
+import akka.http.scaladsl.model.HttpRequest
+import castalia.matcher.MatcherActor.RespondIfMatched
+import castalia.matcher.RequestMatcherActor.{AddMatcher, FindMatchAndForward}
+import castalia.matcher.types._
+
+/**
+  * Created by m06f791 on 4-2-2016.
+  */
+
+object RequestMatcherActor {
+
+  case class FindMatchAndForward(httpRequest: HttpRequest, origin: ActorRef)
+
+  case class AddMatcher(segments: Segments, handler: ActorRef)
+
+  def props(): Props = Props(new RequestMatcherActor with RequestMatcherActorCreatorImplementation)
+}
+
+class RequestMatcherActor extends Actor with ActorLogging {
+  this: RequestMatcherActorCreator =>
+
+  def receive: Receive = normal(Set.empty[(ActorRef, Segments)])
+
+  def normal(matchers: Set[(ActorRef, Segments)]): Receive = {
+    // Add matcher. If there is already a matcher with the same segments, stop that and replace with new one
+    case AddMatcher(segments, handler) =>
+      val matcher = createRequestMatcherActor(context, segments, handler)
+      log.debug(s"Added matcher $matcher to RequestMatcherActor ${self.toString()}")
+      matchers.find(_._2.equals(segments)).map{case (matcherRef, _) => context.stop(matcherRef)}
```

Comment:
IntelliJ: Expression with unit return type in the argument of map. Detects using map with expressions which produce no result. If possible, this inspection suggest to replace such map call with foreach one.